test/integration/runner-spec/iterationCount.test.js

```diff
@@ -0,0 +1,230 @@
+var collection = {
+    item: [{
+        event: [{
+            listen: 'prerequest',
+            script: {
+                exec: `
+                    pm.test('should contain data', function () {
+                        pm.expect(pm.variables.get('foo')).to.equal('bar');
+                    });
+                `
+            }
+        }],
+        request: {
+            url: 'https://postman-echo.com/get',
+            method: 'GET'
+        }
+    }]
+};
+
+describe('Run option iterationCount', function () {
+    describe('when set', function () {
+        describe('without data', function () {
+            var testrun;
+
+            before(function (done) {
+                this.run({
+                    iterationCount: 3,
+                    collection
+                }, function (err, results) {
+                    testrun = results;
+                    done(err);
+                });
+            });
+
+            it('should run collection specified number of times', function () {
+                expect(testrun).to.be.ok;
+                expect(testrun.done.getCall(0).args[0]).to.not.exist;
+                expect(testrun).to.nested.include({
+                    'done.calledOnce': true,
+                    'start.calledOnce': true
+                });
+
+                expect(testrun.request.callCount).to.equal(3);
+            });
+        });
+
+        describe('with data items lesser than iteration count', function () {
+            var testrun;
+
+            before(function (done) {
+                this.run({
+                    iterationCount: 4,
+                    data: [
+                        {foo: 'bar'},
+                        undefined,
+                        {foo: 'bar'}
+                    ],
+                    collection
+                }, function (err, results) {
+                    testrun = results;
+                    done(err);
+                });
+            });
+
+            it('should complete the run successfully', function () {
+                expect(testrun).to.be.ok;
+                expect(testrun.done.getCall(0).args[0]).to.not.exist;
+                expect(testrun).to.nested.include({
+                    'done.calledOnce': true,
+                    'start.calledOnce': true
+                });
+            });
+
+            it('should use last data value for iterations with no corresponding data', function () {
```

Comment:
Only assert for the last iteration here

Reply:
What we're also testing here is that if there is an `undefined` in data value, we don't use the last element of `data`.
See `assertions[1]` test.